### PR TITLE
feat: added support for long text

### LINKED ISSUE #132: Add File Support in Discord Adapter for Long Text
## Description
This issue tracks the implementation of a feature supporting long text messages on Discord. Currently, the discord adapter is not able to send long text due to restrictions. Although, this can be circumvented by using files. The requirement is to add support for sending long text into files in discord adapter.

### 1. Go through documentation
- **Task**: Checkout what is the length of text accepted by Discord for which it can render the message.
- **Approval**: Get the code reviewed by providing code snippet on how you will be handling this check.

### 2. Implement feature
- **Task**: Implement the above approved feature.

### 3. Error handling
- **Task**: Add error handling to the adapter
- **Details**: Ensure that the feature is thoroughly tested and and have error handling in case of failure.

## Expected Outcomes
- Have a Discord adapter which sends message on a channel, text or file, based on the length of message provided.
- Error handling in case of failure to send message to discord.

## Product Name
packages

## Project Name
packages

## Organization Name
SamagraX

## Domain
NA

## Tech Skills Needed
Node JS

## Category
Feature

## Mentor(s)
@ChakshuGautam 

## Complexity
Low

### PR DESCRIPTION
**Pull request by 1010varun**
resolves #132 

## Summary by CodeRabbit

- **Refactor**
  - Improved error handling for server response failures.
  - Updated test setup with mocking for better test isolation.

- **New Features**
  - Enhanced message-sending functionality to handle messages over 2000 characters by uploading files.
  - Added tests for handling messages over 2000 characters and verifying error responses.

### CODE REVIEW

**Comment by 1010varun** [comment]
For short messages
image

if the message is long then it will be shared as a file
image

File: packages/adapters/discord/src/lib/discord.provider.ts [hunk 1]
```diff
@@ -3,28 +3,49 @@ import {
...
+    fs.writeFileSync(tempFilePath, data.content, "utf-8");
+    url.searchParams.set("wait", "true");
+    const form = new FormData();
+    form.append("file", fs.createReadStream(tempFilePath));
+    const response: AxiosResponse<any> = await this.axiosInstance.post(url.toString(),form, {
+        headers: {
+          ...form.getHeaders(),
+        },
+      });
+    fs.unlinkSync(tempFilePath);
```
Comment (chinmoy12c) [inline packages/adapters/discord/src/lib/discord.provider.ts:45]: If request fails, temporary file will not be cleaned up.

File: packages/adapters/discord/src/lib/discord.provider.ts [hunk 2]
```diff
@@ -3,28 +3,49 @@ import {
...
     return {
-      id: response.data.id,
-      date: response.data.timestamp,
+      id: response.data.id,
```
Comment (chinmoy12c) [inline packages/adapters/discord/src/lib/discord.provider.ts:47]: nit: tailing whitespace

File: packages/adapters/discord/src/lib/discord.provider.spec.ts [hunk 3]
```diff
@@ -1,27 +1,26 @@
-import { DiscordProvider } from './discord.provider';
+import { DiscordProvider } from "./discord.provider";
 
-describe("Discord adapter tests", async () => {
-  it('should trigger Discord provider correctly', async () => {
+describe("Discord adapter tests", () => {
+  it("should trigger Discord provider correctly", async () => {
     const provider = new DiscordProvider({
```
Comment (chinmoy12c) [inline packages/adapters/discord/src/lib/discord.provider.spec.ts:5]: can we have a test to check the sanity of actual `sendMessage` function as well? It should check both the cases for text message and file upload and mock the web call.

**Review by chinmoy12c — CHANGES_REQUESTED** [review]

**Comment by 1010varun** [comment]
@chinmoy12c done with the reviewed changes, kindly review it once and suggest any other changes if needed.

**Review by chinmoy12c — APPROVED** [review]
Great Job 👍

**Comment by VedantKhairnar** [comment]
Hello @1010varun   ,
Thanks for your contribution.
As part of Augtoberfest, we want to reward you for your contributions, so pls fill out [this form](https://forms.gle/VBDUPWMYcZEbmVrQ9) by 19th Sept.
Thanks.

**Comment by 1010varun** [comment]
Hey @VedantKhairnar, I saw your comment and filled out the Google form. Apologies for the delay

**Pull request merged**